kafka-elasticsearch/src/main/java/com/elte/elasticSearchConsumer/ElasticSearchAllData.java

```diff
@@ -0,0 +1,121 @@
+    public static RestHighLevelClient createClient(){
+
+        Properties esConfig = new Properties();
+        String esConfigFile = "C:\\Users\\suchi\\IdeaProjects\\OSTproject\\AnomalyDetection\\kafka-elasticsearch\\src\\main\\java\\com\\elte\\elasticSearchConsumer\\elasticsearch.properties";
+        InputStream is = null;
+        try {
+            is = new FileInputStream(esConfigFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        try {
+            esConfig.load(is);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        String hostname = esConfig.getProperty("hostname");
+        String username = esConfig.getProperty("username");
+        String password = esConfig.getProperty("password");
+
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY,
+                new UsernamePasswordCredentials(username, password));
+
+        RestClientBuilder builder = RestClient.builder(
+                new HttpHost(hostname, 443, "https"))
+                .setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
+                    @Override
+                    public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpAsyncClientBuilder) {
+                        return httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                    }
+                });
+
+        RestHighLevelClient client = new RestHighLevelClient(builder);
+        return client;
+    }
+
+    public static KafkaConsumer<String, String> createConsumer(String topic){
+        String bootstrapServers = "127.0.0.1:9092";
+        String groupId = "es-all-balance";
+
+        //create producer properties
+        Properties properties = new Properties();
+        properties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        properties.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        properties.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,StringDeserializer.class.getName());
+        properties.setProperty(ConsumerConfig.GROUP_ID_CONFIG,groupId);
+        properties.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        //create producer
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<String, String>(properties);
+        consumer.subscribe(Arrays.asList(topic));
+
+        return consumer;
+
+    }
+
+    public static void main(String[] args) throws IOException {
+
+        Logger logger = LoggerFactory.getLogger(ElasticSearchAllData.class.getName());
+        RestHighLevelClient client = createClient();
+
+        KafkaConsumer<String, String> consumer = createConsumer("balance_topic");
+        KafkaConsumer<String, String> consumer2 = createConsumer("filtered_balance_topic");
```

Comment:
Is consumer for filtered_balance_topic required here? Why?

Reply:
no, I missed deleting that line. I have created a separate class to consume "filtered_balance_topic". I will make the changes and update again